packages/Liquid-Core.package/LiquidAnswerPanel.class/instance/question..st

```diff
@@ -0,0 +1,14 @@
+accessing
+question: aQuestion
+
+	question := aQuestion.
+	selectedChoices := Set new.
+	(question type = 'singleChoice')
```

Comment:
Also, `question type` seems like it should be encapsulated in class hierarchy instead of a type indicator

Reply:
About `maxNumberOfChoices`: I don't think this would make sense on the Question object, because e.g. a FreeTextQuestion wouldn't have this parameter.
The initial reasoning behind differentiating between `SingleChoiceQuestion` and `MultipleChoiceQuestion` was that I thought we would differentiate between these two types of questions all the time in the UI and it would therefore be easier to just make a new question type for this special case.
If you look in the Domain Model you even see the following comment about this:

> It is possible to represent SingleChoice and MultipleChoice with PrioritisedChoices -> extra class or same data structure and a bit more logic in the ui? The latter would even allow for conversion between these types.

At the end we just implemented the structure that was in the initial draft of the DomainModel.
Because things went different than I originally expected we now have more work, because of the additional question type. Therefore I would agree with Nikkels proposal to just treat SingleChoiceQuestion as a special type of multipleChoiceQuestion. But this should be another PR.

> Also, question type seems like it should be encapsulated in class hierarchy instead of a type indicator

I originally thought we could not "send" class information or functions over the network. So to differentiate between the types of the questions you would need a `type` field. We could now also do this via `kindOf` or `instanceOf` or sth. like this I guess. Solving this solely via polymorphism wouldn't work, because e.g. `FreeTextQuestion` has completely different relevant properties in contrast to a `Choice`-question

Reply:
> a FreeTextQuestion wouldn't have this parameter.

a FreeTextQuestion wouldn't be accessed from this code anyway, right?

>  treat SingleChoiceQuestion as a special type of multipleChoiceQuestion. But this should be another PR.

👍 times two

> Solving this solely via polymorphism wouldn't work, because e.g. FreeTextQuestion has completely different relevant properties

both have an `id`, both are linked to a `PollDraft`, both can be related to `AnswerSet`. Only because their interfaces aren't identical doesn't mean they can't stem from a common ancestor.

Reply:
My understanding/vision for this is as follows:
The class we are talking about here is the `LiquidAnswerPanel` - It gets a Question and provides the user with the ability to answer it.
Because we only have "Choice"-Questions at the moment the ability to answer a question is covered by providing a (Multi)SelectionList and is therefore currently in this PR already integrated in the `LiquidAnswerPanel`.
If we have `FreeTextQuestions` too, we would probably make a `ChoiceAnswerPanel` (= MultiSelectionList like now) and a `FreeTextAnswerPanel` (= a simple textarea) and display the right one according to the type of the question.


> a FreeTextQuestion wouldn't be accessed from this code anyway, right?

It would actually - either in the same `LiquidAnswerPanel` or in a child of it.

> both have an `id`, both are linked to a `PollDraft`, both can be related to `AnswerSet`. Only because their interfaces aren't identical doesn't mean they can't stem from a common ancestor.

They already have a common parent: `Question`. But the displayed ui depends on the type of the question (selectionList or textarea). Therefore we would need some way to see of what type a question is. - Or is your proposal to put the necessary ui for answering a question in the question itself?

Reply:
@Skn0tt Is this discussion blocking [boulder]?